shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -469,12 +469,11 @@
 bool FlutterPlatformViewsController::SubmitFrame(GrDirectContext* gr_context,
                                                  std::shared_ptr<IOSContext> ios_context,
                                                  std::unique_ptr<SurfaceFrame> frame) {
-  FML_DCHECK(flutter_view_);
-
   // Any UIKit related code has to run on main thread.
-  // When on a non-main thread, we only allow the rest of the method to run if there is no
-  // Pending UIView operations.
-  FML_DCHECK([[NSThread currentThread] isMainThread] || !HasPlatformViewThisOrNextFrame());
+  FML_DCHECK([[NSThread currentThread] isMainThread]);
+  if (flutter_view_ == nullptr) {
```

Comment:
@gaaclarke @blasten This should fix the race

Reply:
This also needs to be in `CompositeEmbeddedView`. Amir had a `TODO` right after the check.

Reply:
Done